corehq/apps/app_manager/tests/views/test_apply_patch.py

```diff
@@ -0,0 +1,103 @@
+import difflib
+
+from ...views.forms import apply_patch
+
+
+def test_empty_patch():
+    diff(apply_patch("", ""), "")
+
+
+def test_vellum_xml_patch():
+    # Form and patch generated by creating a simple form in stand-alone Vellum
+    # and then inspecting a save call in src/core.js with the debugger.
+    diff(apply_patch(PATCH, FORM_XML), PATCHED_FORM)
+
+
+def diff(actual, expected):
```

Comment:
I'm not sure that you need to [implement a custom assertion mixin](https://stackoverflow.com/questions/6655724/how-to-write-a-custom-assertfoo-method-in-python), but what you're doing here is a custom assert. Since methods like `assertSetEqual(...)` exist, I think `assertDiffEqual` is more appropriate than just `diff` here.

Reply:
I originally used assertEqual (actually [`eq`](https://github.com/millerdev/testil/blob/553c49d61a99808f1dcd11f97f4dfa55ad1df634/testil.py#L22)), but it was very hard to find the difference if it failed. Given the nature of the compared content, this helper makes that much easier to see.